pkg/utils/comparators/envvars.go

```diff
@@ -12,77 +12,108 @@ Notes:
 - reflect.DeepEqual does not return expected results if the to-be-compared value is a pointer.
 - needs to adjust with k8s.io/api/core/v#/types.go when the types are updated.
 **/
-func EnvValueEqual(env1, env2 []v1.EnvVar) bool {
-	var found bool
-	if len(env1) != len(env2) {
+func EnvValueEqual(lhs, rhs []v1.EnvVar) bool {
```

Comment:
is this refactor needed here?

Reply:
this function was incorrectly returning "false" and causing the cluster to be rolled out again unnecessarily. this is how i rewrote it